### ISSUE TITLE
RFC: Search architecture

### ISSUE THREAD

**Issue by onedr0p** [zoriya/Kyoo#382]
### Feature description

Hi 👋🏼 

I noticed you are using meilisearch and was wondering if you knew about, or had any plans to switch this architecture to [pgvecto.rs](https://github.com/tensorchord/pgvecto.rs). The main benefit here is the reduction of additional containers while still having fast search abilities. [Immich](https://github.com/immich-app/immich) recently changed from Typesense from pgvecto.rs in [v1.91.0 ](https://github.com/immich-app/immich/releases/tag/v1.91.0). Using pgvecto.rs does require a custom PG container with the extension installed and enabled but that is not hard to build or deploy. tensorchord actually provides a Docker container for this as we can see in the Immich [docker-compose](https://github.com/immich-app/immich/blob/8f981b60529140af839171dc052bc49784c73911/docker/docker-compose.prod.yml#L60-L62).

Thanks!

**Comment by zoriya** [comment]
Can you explain why removing containers is easier for deployment? Since users should just consume docker-compose files or helm charts, I don't really get how that impact the users.

**Comment by bo0tzz** [comment]
For the users that just apply your docker-compose it doesn't make much difference indeed. But many people deploy through different methods or on different platforms, or edit compose files to add things like traefik annotations and such. In those cases, people need to build a better understanding of what each container does and how they interact.

That said, I'm on board with an approach of "if you don't use our compose file, you'll need to figure it out yourself". I think the development story is a stronger argument. For example, when we used typesense for Immich it meant there was a whole separate database that we had to keep synchronised with postgres. This was a huge pain, full of bugs, and accounted for something like 20% of the lines of code in the projects. Removing it has been one of the better choices we've made :)

**Comment by zoriya** [comment]
Yeah I can see that, meilisearch syncing was done quickly tho and is ~400loc for the whole feature.
I'd like to train myself in k8s/devops things, so I want to write a helm chart that would handle most of the common scenarios (traekic, graphana...) but I must say I'm a big newbie in this area for new x)

**Comment by joryirving** [comment]
I'm hoping to deploy this shortly via helm chart using a generic chart such as [app-template](https://bjw-s.github.io/helm-charts/docs/common-library/)
